Lesson 4

ignatbagaev opened the pull request.

ignatbagaev committed:
make user to be able to create questions

ignatbagaev committed:
make user to be able to create answers

vkurennov commented on app/controllers/answers_controller.rb:
```diff
@@ -0,0 +1,26 @@
+  end
+
+  def create
+    @answer = Answer.new answer_params.merge(question: @question)
```
Связанные объекты лучше создавать через ассоциации: @question.answers.new

vkurennov commented on spec/controllers/answers_controller_spec.rb:
```diff
@@ -0,0 +1,47 @@
+      it 'creates new answer in database' do
+        expect { post :create, question_id: question.id,
+                               answer: attributes_for(:answer) 
+                }.to change(Answer, :count).by 1
```
Нужно проверить, что созданный ответ связан с вопросом. Это можно сделать прямо в change через ассоциации

vkurennov commented:
Исправь замечания

ignatbagaev committed:
fix solution of lesson 4

Changed the way of Answer creating and added one more spec to test that newly created answer is associated with certain question

palkan commented:
Принято
